src/Orleans.TestingHost/SiloHandle.cs

```diff
@@ -10,24 +8,24 @@ namespace Orleans.TestingHost
     /// </summary>
     public abstract class SiloHandle : IDisposable
     {
-        /// <summary> Get or set configuration of the silo </summary>
-        public NodeConfiguration NodeConfiguration { get; set; }
+        /// <summary> Get or set configuration of the cluster </summary>
+        public TestClusterOptions ClusterOptions { get; set; }
+
+        /// <summary> Gets or sets the instance number within the cluster.</summary>
+        public short InstanceNumber { get; set; }
 
         /// <summary> Get or set the name of the silo </summary>
         public string Name { get; set; }
 
         /// <summary>Get or set the address of the silo</summary>
         public SiloAddress SiloAddress { get; set; }
 
-        /// <summary>Get the proxy address of the silo</summary>
-        public SiloAddress ProxyAddress => SiloAddress.New(this.NodeConfiguration.ProxyGatewayEndpoint, 0);
+        ///// <summary>Get the proxy address of the silo</summary>
+        public SiloAddress GatewayAddress { get; set; }
```

Comment:
the default implementation *I think* could just look for this information in the TestClusterOptions if it uses test cluster membership.

This way other subclasses do not have to rely on extended RPC for it to work (such as MarshalByRefObject as we have in the AppDomainSiloHandle implementation)

Reply:
I wanted this so that I could query it for some tests. It seems right that if SiloAddress is there, then GatewayAddress should also be there.

On the other hand, I'm not confident about the overall class structure for this change (i.e, could things be separated more cleanly?)

Reply:
Sure, it's fine, I don't really mind if we keep this way for now. The reality is that I initially was retrieving it from the container in that way before I did all the work to move to strongly typed config. Today you can just as well get it directly from the configuration.
But as I said, it's fine... Both AppDomain and in-domain strategies can RPC from the TestCluster... If we have other approaches (like out of process or containers) we can refactor it so that the silo handle infers it from the config instead of asking the remote host.